examples/etherpad/Nulecule.yaml

```diff
@@ -0,0 +1,104 @@
+# Using etherpad as an example
+---
+metadata:
+  name: Etherpad
+  id: etherpad
+  appversion: 0.0.1
+  spec: 1.0.0
+  description: Etherpad lite
+
+provider:
+  default: kubernetes
+  secrets: # Encrypted on build
+    provider-api: https://localhost:8080
+    provider-auth: foobar
+    provider-tlsverify: true
+    provider-cacert: |
+      ----BEGIN CERTIFICATE REQUEST-----
+      FOO
+      -----END CERTIFICATE REQUEST-----
+
+# Global parameters
+params: 
+  db_user:
+    description: Database username
+  db_pass:
+    description: Database password
+  db_name:
+    description: Database name
+  db_host:
+    description: Database service name
+    default: mariadb
+ 
+# Inheriting an application
+db:
+  source: docker://projectatomic/mariadb:latest
+
+etherpad:
+
+  params: 
+    image:
+      description: Container image
+      default: centos/etherpad
+    hostport:
+      description: Host port used
+      default: 9001
+  
+  provider: 
+    docker:
+      - run: | # Variable can be whatever name you wish. May in future add "hooks" (before-perform, after-perform, etc.)
+        docker run --link=mariadb-atomicapp-app:db -d --name etherpad -p $hostport:9001 $image
+
+    kubernetes:
+      - rc: |
+        ---
+        apiVersion: v1
+        kind: ReplicationController
+        metadata:
+          name: etherpad
+          labels:
+            name: etherpad
+        spec:
+          replicas: 1
+          selector:
+            name: etherpad
+          template:
+            metadata:
+              labels:
+                name: etherpad
+            spec:
+              containers:
+                - name: etherpad
+                  image: $image
+                  ports:
+                    - name: etherpad-server
+                      containerPort: $hostport
+                  env:
+                    - name: DB_HOST
+                      value: $db_host
+                    - name: DB_DBID
+                      value: $db_name
+                    - name: DB_PASS
+                      value: $db_pass
+                    - name: DB_PORT
+                      value: '3306'
+                    - name: DB_USER
+                      value: $db_user
```

Comment:
Yeah! It's just piping in the file. The example I used had the artifacts as .yaml so that's why it looks messy. Have a look at the helloapache example below :)